analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -215,7 +215,12 @@ variableDeclarator.Parent.Parent switch
                 };
 
             Diagnostic CreateS1144Diagnostic(SyntaxNode syntaxNode, ISymbol symbol) =>
-                Diagnostic.Create(RuleS1144, syntaxNode.GetLocation(), accessibility, symbol.GetClassification(), GetMemberName(symbol));
+                Diagnostic.Create(RuleS1144, GetIdentifierLocation(syntaxNode), accessibility, symbol.GetClassification(), GetMemberName(symbol));
+
+            static Location GetIdentifierLocation(SyntaxNode syntaxNode) =>
+                syntaxNode.GetIdentifier().HasValue
```

Comment:
I think `GetIdentifier(SyntaxNode)` does not support local functions. Please add a test for local functions if there isn't any and make sure it behaves as expected.

Reply:
I've added the test case. The analyzer doesn't support local functions at all (I also checked it in SonarLint).
That seems to be outside the scope of this PR, so I created a new issue for it: #6699